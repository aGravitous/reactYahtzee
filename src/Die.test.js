import Die from './Die';
import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

// Smoke test for Die
it('renders without crashing', () => {
    shallow(<Die />);
});

// Snapshot test for unlocked die
it("matches unlocked snapshot", function (){
    wrapper = shallow(<Die
        locked={false}
        val={1}
        />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});


// Snapshot test for locked die
it("matches locked snapshot", function (){
    wrapper = shallow(<Die
        locked={true}
        val={1}
        />);
        let serialized = toJson(wrapper);
        console.log(serialized);
    expect(serialized).toMatchSnapshot();
});