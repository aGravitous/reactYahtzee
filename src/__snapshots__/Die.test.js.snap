// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches locked snapshot 1`] = `
<button
  className="Die Die-locked"
  onClick={[Function]}
>
  1
</button>
`;

exports[`matches unlocked snapshot 1`] = `
<button
  className="Die"
  onClick={[Function]}
>
  1
</button>
`;
